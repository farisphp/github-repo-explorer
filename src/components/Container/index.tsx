import React from 'react'
import cn from 'classnames'

function Container({
  children,
  className,
  ...props
}: React.HTMLAttributes<HTMLDivElement>) {
  return (
    <div className={cn('mx-auto px-3 xl:px-0 max-w-5xl', className)} {...props}>
      {children}
    </div>
  )
}

export default Container
