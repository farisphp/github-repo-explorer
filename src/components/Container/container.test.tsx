import { render } from '@testing-library/react'

import Container from '.'

describe('<Container />', () => {
  it('should render with className bg-black', () => {
    const { container } = render(<Container>test</Container>)
    const wrapper = container.getElementsByClassName('max-w-5xl')

    expect(wrapper[0]).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
