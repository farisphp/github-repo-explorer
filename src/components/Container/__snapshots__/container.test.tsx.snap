// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Container /> > should render with className bg-black 1`] = `
<div
  class="mx-auto px-3 xl:px-0 max-w-5xl"
>
  test
</div>
`;
